lms/djangoapps/mobile_api/social_facebook/utils.py

```diff
@@ -64,5 +64,11 @@ def share_with_facebook_friends(friend):
     """
     Return true if the user's share_with_facebook_friends preference is set to true.
     """
-    share_fb_friends_settings = preference_info(friend['edX_username'])
-    return share_fb_friends_settings.get('share_with_facebook_friends', None) == 'True'
+
+    # Calling UserPreference directly because the requesting user may be different (and not is_staff).
```

Comment:
I'm finding this code hard to read. Why does it use filter rather than get? Is it possible to have more than one preference value?